test/error-handling.js

```diff
@@ -143,4 +144,25 @@ describe('pool error handling', function () {
       return pool.end()
     }))
   })
+
+  it('should continue with queued items after a connection failure', (done) => {
+    const closeServer = net.createServer((socket) => {
+      socket.destroy()
+    }).unref()
```

Comment:
is the unref needed even though you call close on line 162?  It's not a big deal, just curious.